doit/task.py

```diff
@@ -367,7 +371,7 @@ def init_options(self):
         if self.options is None:
             taskcmd = TaskParse([CmdOption(opt) for opt in self.params])
             # ignore positional parameters
-            self.options = taskcmd.parse('')[0]
```

Comment:
here is the key point. Before calling `taskcmd.parse` we should call:

```
taskcmd.overwrite_defaults(self.cfg_defaults)
```

Reply:
I've fixed this in both locations. Thanks... I failed to notice `.overwrite_defaults()` was available. Much clearer.